Source/Core/VideoCommon/VertexLoaderManager.cpp

```diff
@@ -321,45 +323,96 @@ void LoadCPReg(u32 sub_cmd, u32 value, bool is_preprocess)
   CPState* state = is_preprocess ? &g_preprocess_cp_state : &g_main_cp_state;
   switch (sub_cmd & CP_COMMAND_MASK)
   {
+  case UNKNOWN_00:
+  case UNKNOWN_10:
+  case UNKNOWN_20:
+    if (sub_cmd != UNKNOWN_20 || value != 0)
+    {
+      // All titles using libogc or the official SDK issue 0x20 with value=0 on startup
+      DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_CP_PERF_COMMAND);
+      DEBUG_LOG_FMT(VIDEO, "Unknown CP command possibly relating to perf queries used: {:02x}",
+                    sub_cmd);
+    }
+    break;
+
   case MATINDEX_A:
+    if (sub_cmd != MATINDEX_A)
+    {
+      DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_MAYBE_INVALID_CP_COMMAND);
+      WARN_LOG_FMT(VIDEO,
+                   "CP MATINDEX_A: Command does not match exactly ({:02x} instead of {:02x})",
+                   sub_cmd, MATINDEX_A);
+    }
+
     if (update_global_state)
       VertexShaderManager::SetTexMatrixChangedA(value);
     break;
 
   case MATINDEX_B:
+    if (sub_cmd != MATINDEX_B)
+    {
+      DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_MAYBE_INVALID_CP_COMMAND);
+      WARN_LOG_FMT(VIDEO,
+                   "CP MATINDEX_B: Command does not match exactly ({:02x} instead of {:02x})",
```

Comment:
I'm gonna be really nitpicky here.  I feel like this is worded oddly.  The expectation and the value given seem backward.  Here's my take:

```
CP MATINDEX_B:  an exact value of {MATINDEX_0} was expected but instead a value of {sub_cmd} was seen
```

Reply:
Changed.  It definitely was oddly worded, but I couldn't think of a better phrasing.